File: app/src/main/java/io/github/zwieback/familyfinance/business/chart/service/formatter/LocalizedValueFormatter.java
package io.github.zwieback.familyfinance.business.chart.service.formatter;

import com.github.mikephil.charting.formatter.ValueFormatter;

import java.math.BigDecimal;

import io.github.zwieback.familyfinance.util.NumberUtils;

public class LocalizedValueFormatter extends ValueFormatter {

    @Override
    public String getFormattedValue(float value) {
        return NumberUtils.bigDecimalToString(BigDecimal.valueOf(value),
                NumberUtils.ACCOUNT_PLACES);
    }
}

File: app/src/main/java/io/github/zwieback/familyfinance/business/chart/service/formatter/LocalizedValueFormatter.kt
package io.github.zwieback.familyfinance.business.chart.service.formatter

import com.github.mikephil.charting.formatter.ValueFormatter
import io.github.zwieback.familyfinance.constant.AccountNumberConstants
import io.github.zwieback.familyfinance.extension.toStringWithPlaces
import java.math.BigDecimal

class LocalizedValueFormatter : ValueFormatter() {

    override fun getFormattedValue(value: Float): String {
        return BigDecimal.valueOf(value.toDouble())
            .toStringWithPlaces(AccountNumberConstants.ACCOUNT_PLACES)
    }
}
